implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxApiVersion.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2017 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.influx;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import io.micrometer.core.instrument.util.StringUtils;
+import io.micrometer.core.ipc.http.HttpSender;
+
+/**
+ * @author Jakub Bednar (15/03/2021 14:08)
+ */
+public enum InfluxApiVersion {
+    V1 {
+        @Override
+        String writeEndpoint(final InfluxConfig config) {
+            String influxEndpoint = config.uri() + "/write?consistency=" + config.consistency().toString().toLowerCase() + "&precision=ms&db=" + config.db();
+            if (StringUtils.isNotBlank(config.retentionPolicy())) {
+                influxEndpoint += "&rp=" + config.retentionPolicy();
+            }
+            return influxEndpoint;
+        }
+
+        @Override
+        void addHeaderToken(final InfluxConfig config, final HttpSender.Request.Builder requestBuilder) {
```

Comment:
I suppose the problem is that if the `token` isn't configured, then a header shouldn't be added. That logic would need to be implemented at the callsite instead of here in that case.